portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java

```diff
@@ -41,16 +58,119 @@ public void uncaughtException(Thread thread, Throwable t) {
           }
         });
 
+    ApiMain apiMain = new ApiMain();
+    apiMain.initialize();
+    apiMain.start();
+  }
+
+  public void initialize() {
+    TypeManager typeManager = new TypeManagerImpl();
+
+    // TODO implement
+    Map<String, Object> configuration = new HashMap<>();
+
+    ExtensionContext extensionContext = new ApiExtensionContext(typeManager, configuration);
+
+    ServiceLoader.load(ServiceExtension.class)
+        .iterator()
+        .forEachRemaining(serviceExtension -> serviceExtension.initialize(extensionContext));
+
+    CloudExtension cloudExtension = getCloudExtension();
+    cloudExtension.initialize(extensionContext);
+
     // TODO: Support other server implementations, e.g. Jetty, Tomcat
     // TODO: Don't hardcode list of services
     Set<String> services = ImmutableSet.of("microsoft");
 
+    // FIXME make configurable
+    SymmetricKeyGenerator keyGenerator = new AesSymmetricKeyGenerator();
+
     Injector injector =
         Guice.createInjector(
-            new PortabilityAuthServiceProviderModule(services), new ReferenceApiModule());
+            new ApiServicesModule(typeManager, cloudExtension.getJobStore(), keyGenerator),
```

Comment:
this guice config looks off to me, but if it gets stuff working for the demo we can look into it more later

if you look at WorkerMain that is all configured to use Guice, you should only need to install the ApiModule and everything else should be automagically configured from there..

Reply:
I added the ApiServicesModule to make the instances created at bootstrap available for injection in core services. Is there something that doesn't seem right about that?

Reply:
actually never mind, I forgot that JobStore etc isn't provided by Guice all the time, only for CloudExtensions who choose to use Guice. You can disregard this...